src/server/transaction.cc

```diff
@@ -1129,7 +1129,7 @@ bool Transaction::ScheduleInShard(EngineShard* shard, bool execute_optimistic) {
     DVLOG(3) << "Lock granted " << lock_granted << " for trans " << DebugId();
 
     // Check if we can run immediately
-    if (shard_unlocked && execute_optimistic && lock_granted) {
+    if (lock_granted && execute_optimistic) {
```

Comment:
`lock_granted = shard_unlocked && keys_unlocked;` so checking explicitly for `shard_unlocked` is redundant